common/src/main/java/org/apache/gravitino/dto/requests/CatalogCreateRequest.java

```diff
@@ -79,6 +81,17 @@ public CatalogCreateRequest(
     this.properties = properties;
   }
 
+  /**
+   * Sets the provider of the catalog.
+   *
+   * @param provider The provider of the catalog.
+   */
+  @JsonSetter(value = "provider")
+  public void setProvider(String provider) {
+    this.provider =
+        StringUtils.isNotBlank(provider) ? provider : CatalogProvider.builtinShortName(type);
```

Comment:
I'm afraid that we should not mix up the concept provider and type, in your code, for the built-in catalog, the value of type is the same as that of the provider.

Another point,  in the code above, the provider of the model catalog is `null`, why do we set it to `builtinShortName ` NOT null here.

Reply:
Do you have better suggestion?

Reply:
This makes sense, but it can't cover some cases. For instance, if users set the type = 'relational' and `provider`= 'null', then the final error will be very misleading, and users may have a problem finding what's wrong. 

<img width="1528" alt="image" src="https://github.com/user-attachments/assets/8e18e1b6-8a12-471f-9c0f-c4680c82ba64">

Reply:
I see. I will add checks.